DPGAnalysis/SiStripTools/plugins/APVCyclePhaseProducerFromL1ABC.cc

```diff
@@ -192,24 +198,43 @@ void APVCyclePhaseProducerFromL1ABC::produce(edm::Event& iEvent, const edm::Even
     long long orbitoffset = _orbitoffsetSOR;
     int bxoffset = 0;
 
-    for (L1AcceptBunchCrossingCollection::const_iterator l1abc = pIn->begin(); l1abc != pIn->end(); ++l1abc) {
-      if (l1abc->l1AcceptOffset() == 0) {
-        if (l1abc->eventType() != 0) {
-          orbitoffset = (long long)iEvent.orbitNumber() - (long long)l1abc->orbitNumber();
-          bxoffset = iEvent.bunchCrossing() - l1abc->bunchCrossing();
-
-          if (_wantHistos) {
-            if (_hbx && *_hbx)
-              (*_hbx)->Fill(l1abc->bunchCrossing());
-            if (_hdbx && *_hdbx)
-              (*_hdbx)->Fill(bxoffset);
-            if (_hdorbit && *_hdorbit)
-              (*_hdorbit)->Fill(orbitoffset);
-          }
-        } else {
-          edm::LogWarning("L1AcceptBunchCrossingNoType") << "L1AcceptBunchCrossing with no type found: ";
-          for (L1AcceptBunchCrossingCollection::const_iterator debu = pIn->begin(); debu != pIn->end(); ++debu) {
-            edm::LogPrint("L1AcceptBunchCrossingNoType") << *debu;
+    try {
+      orbitoffset = (long long)tcdsRecord.getOrbitNr() - (long long)iEvent.orbitNumber();
+      bxoffset =
+          iEvent.bunchCrossing() -
+          tcdsRecord
+              .getBXID();  // In EventWithHistoryProducerFromL1ABC, it's -1\times this, following the corresponding L1ABC line below. I keep this difference here, because I am not sure if it should be kept or not
+      // If I understand correctly, tcdsRecord has no eventType thing, so the control done for l1abc can be safely skipped
+      if (_wantHistos) {
+        if (_hbx && *_hbx)
+          (*_hbx)->Fill(tcdsRecord.getBXID());
+        if (_hdbx && *_hdbx)
+          (*_hdbx)->Fill(bxoffset);
+        if (_hdorbit && *_hdorbit)
+          (*_hdorbit)->Fill(orbitoffset);
+      }
+    } catch (...) {
```

Comment:
What throws the exception? Can it be modified to check the existence/validity explicitly upfront? (exceptions should be left to exceptional circumstances only, preferably left to framework to catch, and `catch(...)` should be avoided)

(same for all other try-catches)

Reply:
same comment as https://github.com/cms-sw/cmssw/pull/29116#discussion_r388455950

Reply:
Right, we crossed.

Reply:
Fixed (will push in a while)